internal/installercache/installercache.go

```diff
@@ -41,53 +48,103 @@ func (fi *fileInfo) Compare(other *fileInfo) bool {
 	return fi.info.ModTime().Unix() < other.info.ModTime().Unix()
 }
 
+const (
+	metricEventInstallerCacheRelease = "installercache.release.metrics"
+)
+
 type Release struct {
-	Path string
+	Path          string
+	eventsHandler eventsapi.Handler
+	// enableInstallerCacheEvent should be set to `true` to send metrics upon release, otherwise `false`.
+	enableInstallerCacheEvent bool
+	// startTime is the time at which the request was made to fetch the release.
+	startTime time.Time
+	// clusterID is the UUID of the cluster for which the release is being fetched.
+	clusterID strfmt.UUID
+	// releaseId is the release that is being fetched, for example "4.10.67-x86_64".
+	releaseID string
+	// cached is `true` if the release was found in the cache, otherwise `false`.
+	cached bool
+	// extractDuration is the amount of time taken to perform extraction, zero if no extraction took place.
+	extractDuration float64
 }
 
-func (rl *Release) Release() {
+// Release is called to signal that the caller has finished using the relase and that resources may be released.
+func (rl *Release) Release(ctx context.Context) {
 	if err := os.Remove(rl.Path); err != nil {
 		logrus.New().WithError(err).Errorf("Failed to delete release link %s", rl.Path)
 	}
+	rl.sendReleaseMetricEvent(ctx)
+}
+
+func (rl *Release) sendReleaseMetricEvent(ctx context.Context) {
+	if !rl.enableInstallerCacheEvent {
+		return
+	}
+	rl.eventsHandler.V2AddMetricsEvent(
+		ctx, &rl.clusterID, nil, nil, "", models.EventSeverityInfo,
+		metricEventInstallerCacheRelease,
+		time.Now(),
+		"release-id", rl.releaseID,
+		"start-time", rl.startTime.Format(time.RFC3339),
+		"cached", rl.cached,
+		"extract-duration", rl.extractDuration,
```

Comment:
Why start-time and extract-duration? Would we need to work out from event-time the total release process duration? Wouldn't an end date help, especially if we need to calculate which releases are concurrent?

Reply:
It feels counter intuitive in a way but the end time is actually the event time. 
We send the event as the very last action.

So start time represents the time the event was requested (via GET) 
event time (on the event itself) is the end time.

Reply:
I see that, do you think that somebody discovering this event (either via kibana or ccx data) would find it helpful to have and endTime, or do you think it can be quickly figured out that event_time is infact endTime as well?

Reply:
I'll add clear start and end time metrics to the event, this will make it way clearer for CCX users.